front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatConf.ts

```diff
@@ -3,23 +3,48 @@ import type { Dispatch } from 'redux';
 
 import type { PowerRestrictionRange } from 'applications/operationalStudies/consts';
 import { formatAllowances } from 'modules/trainschedule/components/ManageTrainSchedule/Allowances/helpers';
-import { NO_POWER_RESTRICTION } from 'modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector';
 import { setFailure } from 'reducers/main';
 import type { OsrdConfState } from 'reducers/osrdconf/consts';
+import { replaceElementAtIndex } from 'utils/array';
 import { kmhToMs } from 'utils/physics';
 import { time2sec } from 'utils/timeManipulation';
 
+// When two consecutive segments have the same power restriction code, they are merged into one. This prevents the back from returning an error.
+function mergeIdenticalNeighbours(powerRestrictionRanges: PowerRestrictionRange[]) {
+  return powerRestrictionRanges.reduce(
+    (mergedArray: PowerRestrictionRange[], currentRange, index, array) => {
+      if (index === 0 || currentRange.value !== array[index - 1].value) {
+        return mergedArray.concat(currentRange);
+      }
+
+      const prevRange = mergedArray[mergedArray.length - 1];
+
+      const mergedRange = {
+        begin: prevRange.begin,
+        end: currentRange.end,
+        value: currentRange.value,
+      };
+
+      const newPreviousRange = {
+        ...prevRange,
+        end: mergedRange.end,
+      };
```

Comment:
You can simplify like this
```suggestion
      const previousRange = mergedArray[mergedArray.length - 1];
      const newPreviousRange = {
        ...prevRange,
        end: currentRange.end,
      };
```

Reply:
fixed